lib/PhysicalobjectCsvImporter.class.php

```diff
@@ -0,0 +1,655 @@
+        $this->setOffset($value);
+
+        break;
+
+      case 'updateSearchIndex':
+        $this->setUpdateSearchIndex($value);
+
+        break;
+
+      default:
+        $this->options[$name] = $value;
+    }
+  }
+
+  public function getOption(String $name)
+  {
+    if (isset($this->options[$name]))
+    {
+      return $this->options[$name];
+    }
+    elseif ('sourceName' == $name)
+    {
+      return basename($this->filename);
+    }
+  }
+
+  public function setUpdateSearchIndex($value)
+  {
+    $this->updateSearchIndex = (bool) $value;
+  }
+
+  public function getUpdateSearchIndex()
+  {
+    return $this->updateSearchIndex;
+  }
+
+  public function setOffset(int $value)
+  {
+    $this->offset = $value;
+  }
+
+  public function getOffset()
+  {
+    return $this->offset;
+  }
+
+  public function setHeader(String $str)
+  {
+    $columnNames = explode(',', trim($str));
+
+    // Trim whitespace
+    $columnNames = array_map('trim', $columnNames);
+
+    // Remove empty values
+    $columnNames = array_filter($columnNames, function ($val) {
+      return !empty($val);
+    });
+
+    if (empty($columnNames))
+    {
+      $msg = <<<EOM
+Invalid header. Please provide a CSV delimited list of column names
+e.g. "name,location,type,culture".
+EOM;
+
+      throw new sfException($msg);
+    }
+
+    // Throw error on unknown column names
+    foreach ($columnNames as $name)
+    {
+      if (!array_key_exists($name, self::$columnMap))
+      {
+        throw new sfException(sprintf('Column name "%s" in header is invalid.',
+          $name));
+      }
+    }
+
+    $this->options['header'] = $columnNames;
+  }
+
+  public function getHeader()
+  {
+    if (isset($this->options['header']))
+    {
+      return $this->options['header'];
+    }
+    else if (null !== $this->reader)
+    {
+      return $this->reader->getHeader();
+    }
+  }
+
+  public function getRow($offset)
+  {
+    $stmt = (new \League\Csv\Statement)->offset($offset);
+
+    return $this->getRecords($stmt)->fetchOne();
+  }
+
+  public function countRowsImported()
+  {
+    return $this->rowsImported;
+  }
+
+  public function countRowsTotal()
+  {
+    return $this->rowsTotal;
+  }
+
+  public function setProgressFrequency(int $freq)
+  {
+    // Note: $progressFrequency == 0 turns off logging
+    $this->progressFrequency = ($freq > 0) ? $freq : 0;
+  }
+
+  public function getProgressFrequency()
+  {
+    return $this->progressFrequency;
+  }
+
+  public function doImport($filename = null)
+  {
+    if (null !== $filename)
+    {
+      $this->setFilename($filename);
+    }
+
+    if (null === $this->filename)
+    {
+      $msg = <<<EOL
+Please use setFilename(\$filename) or doImport(\$filename) to specify the CSV
+file you wish to import.
+EOL;
+      throw new sfException($msg);
+    }
+
+    $this->reader = $this->readCsvFile($this->filename);
+    $stmt = (new \League\Csv\Statement)->offset($this->offset);
+    $records = $this->getRecords($stmt);
```

Comment:
What is the largest number of rows/csv you've imported?  How was the performance?

Reply:
@sbreker 46 rows.  It was really fast. :P  I agree it would be good to test with a much bigger dataset.

Reply:
Yes - curious about memory consumption